packages/state-transition/src/util/shuffle.ts

```diff
@@ -101,7 +101,7 @@ function innerShuffleList(input: Shuffleable, seed: Bytes32, dir: boolean): void
 
   // Seed is always the first 32 bytes of the hash input, we never have to change this part of the buffer.
   const _seed = seed;
-  Buffer.from(_seed).copy(buf, 0, 0, _SHUFFLE_H_SEED_SIZE);
+  buf.set(_seed.subarray(0, _SHUFFLE_H_SEED_SIZE), 0);
```

Comment:
Do we know if there's some perf impact?

Reply:
good call, I further simplified this and added some perf tests that show that .set is roughly equivalent to .copy when there's no allocation of a source subarray.